### PR TITLE
add installation docs

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -66,6 +66,14 @@ flow of data.  Typically people rely on tools like ``itertools.tee``, and
 However this quickly become cumbersome, especially when building complex
 pipelines.
 
+Installation
+------------
+
+To install either use:
+
+- conda-forge: ``conda install streamz -c conda-forge``
+- pip: ``python -m pip install streazm``
```
Comment (martindurant) [inline docs/source/index.rst]: typo here; also, I prefer `pip install astreamz` alone.

File: docs/source/index.rst [hunk 2]
```diff
@@ -66,6 +66,14 @@ flow of data.  Typically people rely on tools like ``itertools.tee``, and
...
+- local pip: ``git clone git@github.com:python-streamz/streamz.git; python -m pip install -e streamz``
```
Comment (martindurant) [inline docs/source/index.rst]: `pip install git+https://github.com/python-streamz/streamz` ?

Comment (CJ-Wright) [reply docs/source/index.rst]: Does that install a local editable copy? (I'm mostly thinking of future streamz developers here)

File: docs/source/index.rst (hunk 1 above)
Comment (CJ-Wright) [reply docs/source/index.rst]: Does that run into issues with the wrong pip being used?

File: docs/source/index.rst (hunk 2 above)
Comment (martindurant) [reply docs/source/index.rst]: No, it uses a temporary copy.
Instead, then, I would say
```
- dev: git clone https://github.com/python-streamz/streamz
       pip install -e streamz/
```

File: docs/source/index.rst (hunk 1 above)
Comment (martindurant) [reply docs/source/index.rst]: Not in my experience. If people have pip on their path from a different environment than python, then I suspect other things would go wrong.

**Comment by CJ-Wright** [comment]
@martindurant now I think this is ready (it also fixes some networkx bugs)

**Comment by CJ-Wright** [comment]
Thank you!

**Pull request merged**